dsgrid/config/project_config.py

```diff
@@ -289,6 +306,143 @@ def dimension_mapping_manager(self):
     def dimension_mapping_manager(self, val: DimensionMappingRegistryManager):
         self._dimension_mapping_mgr = val
 
+    def get_base_dimension(self, dimension: DimensionType):
+        """Return the base dimension matching dimension_type.
+
+        Parameters
+        ----------
+        dimension_type : DimensionType
+
+        Returns
+        -------
+        DimensionConfig
+
+        """
+        for key, dim_config in self.base_dimensions.items():
+            if key.type == dimension:
+                return dim_config
+        assert False, dimension
+
+    def get_supplemental_dimensions(self, dimension: DimensionType):
+        """Return the supplemental dimensions matching dimension (if any).
+
+        Parameters
+        ----------
+        dimension_type : DimensionType
+
+        Returns
+        -------
+        DimensionConfig
+
+        """
+        return [v for k, v in self.supplemental_dimensions.items() if k.type == dimension]
+
+    def get_base_to_base_dimension_mapping_by_types(
+        self, from_dimension: DimensionType, to_dimension: DimensionType
+    ):
+        """Return the base-to-base dimension mapping matching the dimension types.
```

Comment:
Are we calling these association mappings now (instead of base-to-base mappings)? (Maybe related to @daniel-thom 's question to @mooneyme about a possibly defunct-base-to-base field?)

Reply:
Here is my interpretation of things:
- A `dimension mapping` maps a `from_id` to a `to_id`. Example: dataset calls a dimension record something slightly different than the project.
- A `dimension association` in the project.toml specifies what combinations of records are required to be present in the load data table. This association can have two or more columns. If the dataset does have a given combination then it must still have a row for the combination, but the ID is null.

I saw that @mooneyme deleted base-to-base mappings in the project files and am assuming that those are now obsolete.

Reply:
A project is a combination of datasets minus data it does not want. If we have to generate a `dimension association` from datasets anyway, then I think it's easier and safer for project to specify what it does not want from each dataset and have the association be generated based on that.

Reply:
That is right Dan.